commerce/domain/orchestrators/classes/CartCalculateSample.cls

```diff
@@ -3,26 +3,27 @@
 // for orchestrators and then the extension must be linked to the webstore via the appropriate setup.
 
 // This class must extend the CartExtension.CartCalculate class
-public class CartCalculateSample extends CartExtension.CartCalculate {
+global class CartCalculateSample extends CartExtension.CartCalculate {
 
     // Each orchestrator must override this calculate method.
-    public virtual override void calculate(CartExtension.CartCalculateOrchestratorRequest request) {
+    global virtual override void calculate(CartExtension.CartCalculateOrchestratorRequest request) {
         CartExtension.Cart cart = request.getCart();
         CartExtension.CartStatusEnum cartStatus = cart.getStatus();
         CartExtension.CartValidationOutputList cvoList  = cart.getCartValidationOutputs();
+        CartExtension.OptionalBuyerActionDetails optionalBuyerActionDetails = request.getOptionalBuyerActionDetails();
 
         // We might want to determine which calculator we call based on the state of the cart.
         // This block is executed when starting or modifying checkout.
         if (cartStatus == CartExtension.CartStatusEnum.CHECKOUT) {
             // Order matters. If the promotions calculator returns an error, we will finish execution early instead of
             // calling the other calculators
-            promotions(new CartExtension.CartCalculateCalculatorRequest(cart));
+            promotions(new CartExtension.CartCalculateCalculatorRequest(cart, optionalBuyerActionDetails));
```

Comment:
Can you just construct it once in one place above this if block and then pass it to every calculator?